qiskit/circuit/measure.py

```diff
@@ -59,3 +60,80 @@ def measure(self, qubit, cbit):
 
 
 QuantumCircuit.measure = measure
+
+
+class PauliMeasure(Instruction):
+    """Perform a measurement with preceding basis change operations."""
+
+    def __init__(self, basis):
+        """Create a new basis transformation measurement.
+
+        Args:
+            basis (str): The target measurement basis, can be 'X', 'Y' or 'Z'.
+
+        Raises:
+            ValueError: If an unsupported basis is specified.
+        """
+        super().__init__('measure_' + basis, len(basis), len(basis), [])
+
+        transformations = []
+
+        from qiskit.extensions.standard.h import HGate
+        from qiskit.extensions.standard.s import SdgGate
+
+        for qubit_basis in basis:
+            if qubit_basis.lower() == 'x':
+                pre_rotation = post_rotation = [HGate()]
+            elif qubit_basis.lower() == 'y':
+                # since measure and S commute, S and Sdg cancel each other
+                pre_rotation = [SdgGate(), HGate()]
+                post_rotation = [HGate(), SdgGate()]
```

Comment:
should this be Sdg or S? post rotation should be the inverse of the pre rotation.